query-engine/connectors/sql-query-connector/src/ordering.rs

```diff
@@ -125,23 +125,26 @@ pub fn compute_joins_aggregation(
         .path
         .split_last()
         .expect("An order by relation aggregation has to have at least one hop");
+
     let mut joins = rest_hops
         .iter()
-        .map(|rf| compute_one2m_join(base_model, rf, join_prefix.as_str()))
+        .map(|hop| compute_one2m_join(base_model, hop.into_relation_hop().unwrap(), join_prefix.as_str()))
         .collect_vec();
 
     let aggregation_type = match order_by.sort_aggregation {
         SortAggregation::Count => AggregationType::Count,
         _ => unreachable!("Order by relation aggregation other than count are not supported"),
     };
+
     // We perform the aggregation on the last join
     let last_aggr_join = compute_aggr_join(
-        last_hop,
+        last_hop.into_relation_hop().unwrap(),
```

Comment:
I'm assuming this `unwrap` and the next one are safe?

Reply:
Added a comment.